### ISSUE TITLE
DXIL support for Direct3D11

### ISSUE THREAD

**Issue by mrvux** [Microsoft/DirectXShaderCompiler#621]
Hello, 

I was curious about that, so I thought I would post here.

It would be great to also have support for DXIL bytecode in DirectX11 as well.

Since both are still designed to work alongside each other for quite a while, and new Linker features are really promising (and best of all , open source), it would be great that we could be able to have those features for both API (otherwise, would still need to use a different compiler path for DirectX11 and DirectX12, which is a bit annoying).

So my question, is it planned at all (it means to have of course vendors to support it (and having some feature level/profile limitation)?

**Comment by simontaylor81** [comment]
This would be very nice to have indeed! We're currently stuck supporting D3D11 (and likely will be for some time), so we're unable to take advantage of the new features in dxc, even if they have no relevance to the actual API being executed on (e.g. enums).

**Comment by SssWen** [comment]
> Sorry, we currently don't have plans to support DXIL in Direct3D11.

Hello, is there any way to translate DXIL to HLSL?

**Comment by MrTrillian** [comment]
@SssWen No. It's theoretically possible but it would be as complex as writing a decompiler and would produce the same kind of poorly readable code. We have no scenario for it.

**Comment by aras-p** [comment]
Any plans for DXIL -> DXBC translator then? :)

**Comment by Xottab-DUTY** [comment]
There is DXBC -> DXIL translator. If we have that, probably we can have backwards =)

**Comment by aras-p** [comment]
I guess there already must be an DXIL -> DXBC translator for the "DX12 on Windows 7" feature anyway. Would be nice to have that open source and/or at least publicly documented & usable.

**Comment by chloekek** [comment]
It is possible to convert HLSL to SPIR-V using DXC, then convert SPIR-V to HLSL using spirv-cross, and then convert HLSL to DXBC using FXC. This works apart from the conversion being lossy (for instance, structured buffers are turned into byte address buffers), so it may or may not be an acceptable trade-off.
